newIDE/app/src/Profile/ProfileDialog.js

```diff
@@ -45,6 +52,46 @@ const ProfileDialog = ({ currentProject, open, onClose }: Props) => {
     [routeArguments, removeRouteArguments]
   );
 
+  const markBadgesAsSeen = React.useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      if (!(authenticatedUser.authenticated && authenticatedUser.profile)) {
```

Comment:
should you also check the "loginState"?

Reply:
I don't think it's necessary since the achievements component does not display if there's no badges in authenticatedUser.